server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -89,28 +127,50 @@ public void define(Service service) {
             dataType = "String",
             defaultValue = "none",
             example = "user@james.org",
-            value = "Compulsory. Needs to be a valid username represent for an user had requested to restore deleted emails")
+            value = "Compulsory. Needs to be a valid username represent for an user had requested to restore deleted emails"),
+        @ApiImplicitParam(
+            required = true,
+            dataType = "String",
+            name = "action",
+            paramType = "query",
+            example = "?action=restore",
+            value = "Compulsory. Needs to ve a valid action represent for an operation to perform on the Deleted Message Vault")
     })
     @ApiResponses(value = {
         @ApiResponse(code = HttpStatus.CREATED_201, message = "Task is created", response = TaskIdDto.class),
         @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Bad request - user param is invalid"),
         @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
     })
     private TaskIdDto restore(Request request, Response response) {
-        User userToRestore = extractUser(request);
-        TaskId taskId = taskManager.submit(new DeletedMessagesVaultRestoreTask(userToRestore, vaultRestore));
+        Action requestedAction = extractAction(request);
+
+        Task requestedTask = generateTask(requestedAction, request);
+        TaskId taskId = taskManager.submit(requestedTask);
         return TaskIdDto.respond(response, taskId);
     }
 
-    private User extractUser(Request request) {
-        try {
-            return User.fromUsername(request.params(USER_PATH_PARAM));
-        } catch (IllegalArgumentException e) {
-            throw ErrorResponder.builder()
-                .statusCode(HttpStatus.BAD_REQUEST_400)
-                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
-                .message(e.getMessage())
-                .haltError();
+    private Task generateTask(Action requestedAction, Request request) {
+        User userToRestore = User.fromUsername(request.params(USER_PATH_PARAM));
+
+        switch (requestedAction) {
+            case RESTORE:
+                return new DeletedMessagesVaultRestoreTask(userToRestore, vaultRestore);
+            default:
+                throw new IllegalArgumentException("Action " + requestedAction.value + " is not in the supported list" +
+                    "(" + String.join(",", supportedActions()) + ")");
```

Comment:
`Joiner.on(',').join(upportedActions())` is what we commonly used.

Also I propose extracting this to a local `String supportedActions` variables.

Reply:
Do we have some valid actions that are not supported?

Does it makes sense to be covering the same test two time?

Hint: run test coverage tests. You will discover that l 159 + 160 are never run.

I propose to be doing 

```
throw new NotImplementedException(requestedAction + " is not yet handled.");
```

(we are not validating user input but catching future bugs!)